feat(#10959): add release notes for `5.1.2` (#2207)
medic/cht-docs: content/en/releases/5_1_2.md

---
title: "5.1.2 release notes"
linkTitle: "5.1.2"
sidebar:
  exclude: true
description: 
  Released 2026-04-30
---

## Highlights

### Properly trigger user-contact-summary recalculation

CHT `4.21.0` added support for [getting the user's contact summary](/releases/4_21_0/#accessing-the-logged-in-users-contact-summary-in-app-forms-and-app-forms-permissions). Unfortunately, there was a defect in the logic for triggering the recalculation of the user's contact summary that meant the contact summary was not always being recalculated when relevant changes were made. This release fixes that defect, ensuring the user's contact summary is kept up to date as data related to the user's contact changes.

> See [the form inputs documentation](/building/forms/configuring/form-inputs/#user-contact-summary-data) and [the form properties documentation](/building/forms/app/#formsappform_namepropertiesjson) for more details on configuration and functionality.

Reviewer: ```suggestion
See the documentation for [form inputs](/building/forms/configuring/form-inputs/#user-contact-summary-data) and [form properties](/building/forms/app/#formsappform_namepropertiesjson) and also  [#10712](https://github.com/medic/cht-core/issues/10712) for more details.
```

## Known issues

Check the repository for the [latest known issues](https://github.com/medic/cht-core/issues?q=is%3Aissue+label%3A%22Affects%3A+5.1.2%22).

> ## Upgrade notes

Reviewer: ```suggestion
```

### Bug fixes

> - [#10712](https://github.com/medic/cht-core/issues/10712): ContactChangeFilterService.isRelevantChange uses ?? instead of ||, making isRelevantContact and isRelevantReport dead code

Reviewer: ```suggestion
- [#10712](https://github.com/medic/cht-core/issues/10712): ContactChangeFilterService.isRelevantChange uses `??` instead of `||`, making `isRelevantContact` and `isRelevantReport` dead code
```

> - [#10912](https://github.com/medic/cht-core/issues/10912): Authorization context drops duplicate replication-key matches because Set.add() only adds one id

Reviewer: ```suggestion
- [#10912](https://github.com/medic/cht-core/issues/10912): Authorization context drops duplicate replication-key matches because `Set.add()` only adds one ID
```
